avm/res/container-instance/container-group/main.bicep

```diff
@@ -0,0 +1,293 @@
+
+@description('The location the resource was deployed into.')
+output location string = containergroup.location
+
+// =============== //
+//   Definitions   //
+// =============== //
+
+type managedIdentitiesType = {
+  @description('Optional. Enables system assigned managed identity on the resource.')
+  systemAssigned: bool?
+
+  @description('Optional. The resource ID(s) to assign to the resource.')
+  userAssignedResourceIds: string[]?
+}?
+
+type lockType = {
+  @description('Optional. Specify the name of lock.')
+  name: string?
+
+  @description('Optional. Specify the type of lock.')
+  kind: ('CanNotDelete' | 'ReadOnly' | 'None')?
+}?
+
+type customerManagedKeyType = {
+  @description('Required. The resource ID of a key vault to reference a customer managed key for encryption from.')
+  keyVaultResourceId: string
+
+  @description('Required. The name of the customer managed key to use for encryption.')
+  keyName: string
+
+  @description('Optional. The version of the customer managed key to reference for encryption. If not provided, using \'latest\'.')
+  keyVersion: string?
+
+  @description('Optional. User assigned identity to use when fetching the customer managed key. Required if no system assigned identity is available for use.')
+  userAssignedIdentityResourceId: string?
+}?
+
+type containerType = {
+  @description('Required. The name of the container instance.')
+  name: string
+
+  @description('Required. The properties of the container instance.')
+  properties: {
+
+    @description('Required. The name of the container source image.')
+    image: string
+
+    @description('Optional. The liveness probe.')
+    livenessProbe: {}?
+
+    ports: {
+      @description('Required. The port number exposed on the container instance.')
+      port: int
+
+      @description('Required. The protocol associated with the port number.')
+      protocol: string
+    }[]
+
+    resources: {
+      @description('Required. The resource requests of this container instance..')
+      requests: {
+        cpu: int
+        gpu: {
+          count: int
+          sku: string
+        }?
+        memoryInGB: int
+      }
+
+      @description('Optional. The resource limits of this container instance..')
+      limits: {
+        cpu: int
+        gpu: {
+          count: int
+          sku: string
+        }?
+        memoryInGB: string
+      }?
+    }
+
+    @description('Optional. The volume mounts within the container instance.')
+    volumeMounts: {
+      name: string
+      mountPath: string
+    }[]?
+
+    @description('Optional. The command to execute within the container instance.')
+    command: string[]?
+
+    @description('Optional. The environment variables to set in the container instance.')
+    environmentVariables: {
+      name: string
+      secureValue: string?
+      value: string?
```

Comment:
Would be good to have some metadata from the inner parameters to as those are rendered in the readme + are shown by the extension when using the UDTs. The same can be set about all inner properties in each UDT.

Reply:
put in description to each type param